### PR TITLE
Melodic/Noetic support

### PR DESCRIPTION
**Pull request by k-okada**
rework of #39, as #39 mixes style change and noetic support, it also drop meldoic support.

### CODE REVIEW

**Comment by k-okada** [comment]
@knorth55 can you check if  this works on your complex smach example? on both Meldoic / Noetic.

File: smach_viewer/scripts/smach_viewer.py [hunk 1]
```diff
@@ -165,10 +187,10 @@ def update_structure(self, msg):
 
     def _load_local_data(self, msg):
         """Unpack the user data"""
-        try:
-            local_data = pickle.loads(msg.local_data)
-        except KeyError:
-            local_data = pickle.loads(base64.b64decode(msg.local_data.encode('utf-8')))
+        if isinstance(msg.local_data, str):
+            local_data = pickle.loads(base64.b64decode(msg.local_data))
```
Comment (knorth55) [inline smach_viewer/scripts/smach_viewer.py]: this chage causes error in `melodic`.
```
[ERROR] [1658322156.379389]: bad callback: <bound method SmachViewerFrame._status_msg_update of <__main__.SmachViewerFrame; proxy of <Swig Object of type 'wxFrame *' at 0x55a2221c2ce0> >>
Traceback (most recent call last):
  File "/opt/ros/melodic/lib/python2.7/dist-packages/rospy/topics.py", line 750, in _invoke_callback
    cb(msg)
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/scripts/smach_viewer.py", line 879, in _status_msg_update
    if container.update_status(msg):
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/scripts/smach_viewer.py", line 217, in update_status
    self._local_data._data = self._load_local_data(msg)
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/scripts/smach_viewer.py", line 191, in _load_local_data
    local_data = pickle.loads(base64.b64decode(msg.local_data))
  File "/usr/lib/python2.7/base64.py", line 78, in b64decode
    raise TypeError(msg)
TypeError: Incorrect padding
```

**Review by knorth55 — CHANGES_REQUESTED** [review]

File: smach_viewer/scripts/smach_viewer.py (hunk 1 above)
Comment (knorth55) [reply smach_viewer/scripts/smach_viewer.py]: I haven't checked yet, but this may be caused by multibyte language...

Comment (knorth55) [reply smach_viewer/scripts/smach_viewer.py]: Actually it is not related to multibyte language.
we have same error with `roseus_smach/sample/state-machine-ros-sample.l`

Comment (k-okada) [reply smach_viewer/scripts/smach_viewer.py]: I think it is fixed now.

Comment (knorth55) [reply smach_viewer/scripts/smach_viewer.py]: Yes, thank you.

**Comment by knorth55** [comment]
Noetic is not yet tested, but I tested in Melodic.

I tried with english (singlebyte language) and it works in Melodic.
But I merged #40 to test with multibyte japanese but I got the following error.
I don't get this error in #40, so this is a bit weird...

```
[ERROR] [1658325349.225043]: label width 40 causes error
[ERROR] [1658325349.226524]: maybe multibyte word is in your label.
[ERROR] [1658325349.227478]: 'ascii' codec can't decode byte 0xe4 in position 2299: ordinal not in range(128)
[ERROR] [1658325349.228376]: changing width label width to 41
Exception in thread Thread-6:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/knorth55/ros/melodic/src/ros-visualization/executive_smach_visualization/smach_viewer/scripts/smach_viewer.py", line 1090, in _update_graph
    self.widget.subgraph_shapes,
AttributeError: 'WxDotWindow' object has no attribute 'subgraph_shapes'
```

**Comment by k-okada** [comment]
@knorth55 If you revert, you can show  Japanese on melodic
```
-    parser = XDotParser(xdotcode)
+    parser = MyXDotParser(bytes(str(xdotcode).encode("utf-8")))
```

**Comment by knorth55** [comment]
@k-okada 
```
    parser = XDotParser(xdotcode)
```
works.

```
    parser = MyXDotParser(xdotcode)
```
does not works.

So, we should modify `MyXDotParser` to support multibyte language?

**Comment by knorth55** [comment]
@k-okada I make a PR for checking in melodic and noetic https://github.com/k-okada/executive_smach_visualization/pull/1 . Noetic is WIP.

**Pull request merged**